python/src/ops.cpp

```diff
@@ -3035,4 +3035,93 @@ void init_ops(py::module_& m) {
         Returns:
           result (array): The result of the multiplication of ``x`` with ``w``.
       )pbdoc");
+  m.def(
+      "quantize",
+      &quantize,
+      "w"_a,
+      py::pos_only(),
+      "groups"_a = 128,
+      "width"_a = 4,
+      py::kw_only(),
+      "stream"_a = none,
+      R"pbdoc(
+        quantize(w: array, /, groups: int = 128, width: int = 4, *, stream: Union[None, Stream, Device] = None) -> Tuple[array, array, array]
+
+        Quantize the matrix ``w`` using ``width`` bits per element. In
+        particular, every ``groups`` elements in a row of ``w`` are quantized
+        together which means that number of columns of ``w`` should be
```

Comment:
Nit: Can you keep the summary string as a standalone sentence and move the groups to a separate paragraph. Will make the doc autosummaries look better IMO.

So something like:

```
Quantize the matrix ``w`` using ``width`` bits per element.

Note, every ``groups`` elements in a row of ``w`` are quantized
together. Hence, number of columns of ``w`` should be ..
```

In particular, the rows of ``w`` are divided into groups of size ``groups``  which are quantized together.